mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/MailboxIndexerSupplier.java

```diff
@@ -0,0 +1,46 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.elasticsearch;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.james.backends.es.AliasName;
+import org.apache.james.backends.es.DeleteByQueryPerformer;
+import org.apache.james.backends.es.ElasticSearchConstants;
+import org.apache.james.backends.es.Indexer;
+import org.apache.james.backends.es.IndexerSupplier;
+import org.apache.james.backends.es.TypeName;
+import org.elasticsearch.client.Client;
+
+public class MailboxIndexerSupplier implements IndexerSupplier {
```

Comment:
You don't need IndexerSupplier interface at all IMO.
You just bind a specific indexer to a given alias, that can be done in a guice module.

Reply:
Well, a level of indirection seemed the best option to me in order to:

 - Keep the indexer code generic (as it is orthogonal to document structure)
 - But use specific Alias/Mapping

What would be your alternative design? Passing Alias + Mapping as method parameter?